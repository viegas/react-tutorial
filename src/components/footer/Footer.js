import React from 'react';
import PropTypes from 'prop-types';

import styles from './Footer.module.css'

const Footer = ({ size }) => {
    if(!size) {
        return null;
    }

    return (
        <div className={styles.wrapper}>
            <span>{`Items left: ${size}`}</span>
        </div>
    );
};


Footer.propTypes = {
    size: PropTypes.number,
};

Footer.defaultProps = {
    size: 0,
}

export default Footer;
