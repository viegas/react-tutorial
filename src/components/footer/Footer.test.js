import Footer from './Footer';

it('renders correctly when empty', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
});

it('renders correctly with size', () => {
    const size = 1;
    const wrapper = shallow(<Footer size={size} />);

    expect(wrapper.find('span').text()).toBe(`Items left: ${size}`);
    expect(wrapper).toMatchSnapshot();

});
