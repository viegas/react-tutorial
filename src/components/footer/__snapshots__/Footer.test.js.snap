// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when empty 1`] = `""`;

exports[`renders correctly with size 1`] = `
<div
  className="wrapper"
>
  <span>
    Items left: 1
  </span>
</div>
`;
